src/net/relinc/libraries/application/StrikerBar.java

```diff
@@ -33,7 +33,10 @@ public void setSpeed(double speed) {
 	}
 	
 	public boolean isValid(){
-		return !(speed == 0.0 || density == 0.0 || length == 0.0 || diameter == 0.0);
+		return speed > 0.0 || density > 0.0 || length > 0.0 || diameter > 0.0;
```

Comment:
This still is getting used it odd scenarios, like the popover text:
![image](https://user-images.githubusercontent.com/7853605/40146999-c7b78b04-5935-11e8-8947-a4b386a77ea7.png)

Striker Bar Length was never defined by the user, yet it is displayed as `0.0` in the UI.

Same for the "Sample Information Table":
![image](https://user-images.githubusercontent.com/7853605/40147123-5bb2b2ac-5936-11e8-82cb-2d9da2c7824c.png)


Honestly the only place `isValid()` should be used is to decide if the Striker Bar should be saved at all. In all other cases, each property should be handled separately. Unless we are fine with displaying `0.0` to the user (but only if they specify at least one Striker Bar property, else it doesn't display any of the properties)

Reply:
Yep. I'll check for other references in the morning.